app/workers/functions.php

```diff
@@ -168,92 +170,197 @@ function: $function,
+                    '$write' => [],
+                    'dateCreated' => time(),
+                    'functionId' => $functionId,
+                    'deploymentId' => $deploymentId,
+                    'trigger' => $trigger,
+                    'status' => 'waiting',
+                    'statusCode' => 0,
+                    'stdout' => '',
+                    'stderr' => '',
+                    'time' => 0.0,
+                    'search' => implode(' ', [$functionId, $executionId]),
+                ]));
+                
+                if ($execution->isEmpty()) {
+                    throw new Exception('Failed to create or read execution');
+                }
+            }
+            $execution->setAttribute('status', 'processing');
+            $execution = $dbForProject->updateDocument('executions', $executionId, $execution);
+            return $execution;
+        });
+
+        /** Collect environment variables */
+        $vars = [
+            'APPWRITE_FUNCTION_ID' => $functionId,
+            'APPWRITE_FUNCTION_NAME' => $function->getAttribute('name', ''),
+            'APPWRITE_FUNCTION_DEPLOYMENT' => $deploymentId,
+            'APPWRITE_FUNCTION_RUNTIME_NAME' => $runtime['name'],
+            'APPWRITE_FUNCTION_RUNTIME_VERSION' => $runtime['version'],
+            'APPWRITE_FUNCTION_TRIGGER' => $trigger,
+            'APPWRITE_FUNCTION_EVENT' => $event,
+            'APPWRITE_FUNCTION_EVENT_DATA' => $eventData,
+            'APPWRITE_FUNCTION_DATA' => $data,
+            'APPWRITE_FUNCTION_PROJECT_ID' => $projectId,
+            'APPWRITE_FUNCTION_USER_ID' => $userId,
+            'APPWRITE_FUNCTION_JWT' => $jwt,
+        ];
+        $vars = \array_merge($function->getAttribute('vars', []), $vars);
+
+        /** Execute function */
+        $executionResponse = $this->executor->createExecution(
+            projectId: $projectId,
+            functionId: $functionId,
+            deploymentId: $deploymentId,
+            path: $build->getAttribute('outputPath', ''),
+            vars: $vars,
+            entrypoint: $deployment->getAttribute('entrypoint', ''),
+            data: $vars['APPWRITE_FUNCTION_DATA'],
+            runtime: $function->getAttribute('runtime', ''),
+            timeout: $function->getAttribute('timeout', 0),
+            baseImage: $runtime['image']
+        );
+
+        /** Update execution status */
+        $execution->setAttribute('status', $executionResponse['status']);
+        $execution->setAttribute('statusCode', $executionResponse['statusCode']);
+        $execution->setAttribute('stdout', $executionResponse['stdout']);
+        $execution->setAttribute('stderr', $executionResponse['stderr']);
+        $execution->setAttribute('time', $executionResponse['time']);
+        $execution = Authorization::skip(fn() => $dbForProject->updateDocument('executions', $executionId, $execution));
+
+        /** Trigger Webhook */
+        $executionModel = new Execution();
```

Comment:
let's follow similar pattern for naming. Probably the before use of `responseExecute` should be `executionResponse` ?

Reply:
changed both occurrences to `executionResponse ` 👍